### ISSUE TITLE
N+1 API queries in prescription administration table

### ISSUE THREAD

**Issue by rithviknishad** [ohcnetwork/care#2333]
Currently, the code flow in fetching data for the prescription administration table is:

1. List prescriptions for a consultation.
2. Computing the overall start and end administration date range to initialize pagination
3. Compute the latest range required to be shown in view (eg. last X hours)
4. Listing administrations **for each** of the prescriptions within that range.

TODO: A solution to be discussed

**Comment by aeswibon** [comment]
@rithviknishad Is there by solution discussed for this?

**Comment by rithviknishad** [comment]
No @aeswibon 

Feel free to comment a solution

**Comment by gigincg** [comment]
@rithviknishad Can you update here if we are still using the unoptimized logic for Prescription Administration? Reassign to me post update

**Comment by rithviknishad** [comment]
Yes @gigincg We are still using the unoptimized logic to query the prescription administrations in the MAR table (same flow as mentioned in the issue body).

We could consider fetching administrations across prescriptions in 1 API call. But we would need to pass a higher pagination size limit to account for getting all administrations across all prescriptions for the given time range.

<img width="1710" alt="image" src="https://github.com/user-attachments/assets/ee5e1655-1d02-4347-99e8-4c302ac90b7c">
